src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LoadedProject.cs

```diff
@@ -216,6 +219,27 @@ void UpdateProjectSystemProjectCollection<T>(IEnumerable<T> loadedCollection, IE
             if (newItems.Count != oldItemsCount)
                 logger.LogTrace(logMessage, projectFullPathWithTargetFramework, newItems.Count);
         }
+
+        void WatchProjectAssetsFile(ProjectFileInfo currentProjectInfo, IFileChangeContext fileChangeContext)
```

Comment:
I decided to only watch the actual msbuild property for ProjectAssetsFile here - I took a look at a bit of the O# and CPS code, and it looked they were only looking at that specific property (not for any project.assets.json file in obj/).